### ISSUE TITLE
BUG: FlowRouter.reactiveCurrent() causes 2 run, when first time access

### ISSUE THREAD

**Issue by delgermurun** [meteorhacks/flow-router#77]
When access directly (not clicking on anchor), autorun that includes `FlowRouter.reactiveCurrent()`, runs two times. Ideally, it should be run once.

```js
FlowComponents.define('hello', function() {
   this.autorun(function() {
     console.log(FlowRouter.reactiveCurrent());
   });
});
```

**Comment by delgermurun** [comment]
I want to make sure there is an `id` param. Something like that:
```
var comp = Components.define('postDetail', function() {
    this.autorun(function() {
       var id = FlowRouter.getParam('id');
       if (!id) {
          return FlowRouter.go('/');
       }
       console.log(id);
    });
});
```

but first time `id` is `undefined`.

**Comment by elidoran** [comment]
Is the first run when the autorun is registered and the second one when FlowRouter has run and changed the value? 

What if you use `computation.firstRun`? Docs: [computation#firstRun](http://docs.meteor.com/#/full/computation_firstrun).

Do this instead?
```javascript
var comp = Components.define('postDetail', function() {
    this.autorun(function(computation) {
       var id = FlowRouter.getParam('id');
       if (!(computation.firstRun || id)) {
          return FlowRouter.go('/');
       }
       console.log(id);
    });
});```

**Comment by delgermurun** [comment]
@elidoran Thanks for the workaround. It did the trick.
But it feels wrong, I think it should be assigned for first time.

I guess #95 it is also related to it.

**Comment by elidoran** [comment]
I think I found the source of the problem in #95. If approved I'll generate a PR. It's about the order of storing the new values in the `ReactiveDict` in relation to calling the route's `action`.

In your example above you're calling autorun and that is called immediately by FlowComponents [here](https://github.com/meteorhacks/flow-components/blob/master/lib/client/component_instance.js#L11). I haven't used FlowComponents but I briefly looked at its docs and it seems we're supposed to write the components in a script file. So, is the example code you have above in a component script file? If so, then that code is run at app loading time, not in an `action` like #95.

Where is that call to `FlowComponents.define()` being made?
When is the autorun executing and what's triggering it?

**Comment by delgermurun** [comment]
Oh, I forgot to change code. It is exactly same as 
```
var comp = FlowComponents.define('postDetail', function() {
    this.autorun(function() {
       var id = FlowRouter.getParam('id');
       if (!id) {
          return FlowRouter.go('/');
       }
       console.log(id);
    });
});
```

it called when access to page/route.

**Comment by arunoda** [comment]
Released version `1.8.0` with fixes for this bug.

**Comment by delgermurun** [comment]
:+1:
